src/components/register/steps/optional.tsx

```diff
@@ -1,18 +1,25 @@
 /** @jsxImportSource @emotion/react */
 
 import { Localized } from '@fluent/react'
+import { useEffect } from 'react'
 import { RouteComponentProps } from '@reach/router'
 import { useForm } from 'react-hook-form'
 import { Checkbox, FieldSet, TextArea } from '@eurofurence/reg-component-library'
 import WithInvoiceRegisterLayout from '../layout/with-invoice'
 import { OptionalInfo } from '../../../state/models/register'
-import { SubmitOptionalInfo } from '../../../state/actions/register'
+import { ChangeOptionalInfo, SubmitOptionalInfo } from '../../../state/actions/register'
 import { useAppDispatch } from '../../../hooks/redux'
 
 const Optional = (_: RouteComponentProps) => {
-	const { register, handleSubmit } = useForm<OptionalInfo>()
+	const { register, watch, handleSubmit } = useForm<OptionalInfo>()
 	const dispatch = useAppDispatch()
 
+	useEffect(() => {
+		const subscription = watch(data => dispatch(ChangeOptionalInfo.create(data)))
+
+		return () => subscription.unsubscribe()
```

Comment:
Since we need to return a function, can we not just return `subscription.unsubscribe`?